Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -42,80 +42,35 @@
 
 module Homebrew
   module_function
 
-  def inreplace_pairs(path, replacement_pairs)
-    if ARGV.dry_run?
-      contents = path.open("r") { |f| Formulary.ensure_utf8_encoding(f).read }
-      contents.extend(StringInreplaceExtension)
-      replacement_pairs.each do |old, new|
-        unless ARGV.flag?("--quiet")
-          ohai "replace #{old.inspect} with #{new.inspect}"
-        end
-        contents.gsub!(old, new)
-      end
-      unless contents.errors.empty?
-        raise Utils::InreplaceError, path => contents.errors
-      end
-      path.atomic_write(contents) if ARGV.include?("--write")
-      contents
-    else
-      Utils::Inreplace.inreplace(path) do |s|
-        replacement_pairs.each do |old, new|
-          unless ARGV.flag?("--quiet")
-            ohai "replace #{old.inspect} with #{new.inspect}"
-          end
-          s.gsub!(old, new)
-        end
-      end
-      path.open("r") { |f| Formulary.ensure_utf8_encoding(f).read }
-    end
-  end
-
-  def formula_version(formula, spec, contents = nil)
-    name = formula.name
-    path = formula.path
-    if contents
-      Formulary.from_contents(name, path, contents, spec).version
-    else
-      Formulary::FormulaLoader.new(name, path).get_formula(spec).version
-    end
-  end
-
-  def fetch_pull_requests(formula)
-    GitHub.issues_for_formula(formula.name, tap: formula.tap).select do |pr|
-      pr["html_url"].include?("/pull/") &&
-        /(^|\s)#{Regexp.quote(formula.name)}(:|\s|$)/i =~ pr["title"]
-    end
-  rescue GitHub::RateLimitExceededError => e
-    opoo e.message
-    []
-  end
-
-  def check_for_duplicate_pull_requests(formula)
-    pull_requests = fetch_pull_requests(formula)
-    return unless pull_requests
-    return if pull_requests.empty?
-    duplicates_message = <<~EOS
-      These open pull requests may be duplicates:
-      #{pull_requests.map { |pr| "#{pr["title"]} #{pr["html_url"]}" }.join("\n")}
-    EOS
-    error_message = "Duplicate PRs should not be opened. Use --force to override this error."
-    if ARGV.force? && !ARGV.flag?("--quiet")
-      opoo duplicates_message
-    elsif !ARGV.force? && ARGV.flag?("--quiet")
-      odie error_message
-    elsif !ARGV.force?
-      odie <<~EOS
-        #{duplicates_message.chomp}
-        #{error_message}
-      EOS
+  def bump_formula_pr
+    @args = Homebrew::CLI::Parser.parse do
+      switch "--devel"
+      switch "-n", "--dry-run"
+      switch "--write"
+      switch "--audit"
+      switch "--strict"
+      switch "--no-browse"
+      switch "--force"
+      switch :quiet
+      switch :verbose
+      switch :debug
+      flag   "--url",      required: true
+      flag   "--sha256",   required: true
+      flag   "--mirror",   required: true
+      flag   "--tag",      required: true
+      flag   "--revision", required: true
+      flag   "--version",  required: true
+      flag   "--message",  required: true
+      depends :url, :sha256
+      depends :tag, :revision, mandatory: true
```

Comment:
Not clear what `mandatory` means here; how does it differ from the `url` stuff above.

Reply:
`--sha256` is valid only when`--url` is passed, but `--sha256` need not be passed when `--url` is passed. 

`--revision` is valid only when `--tag` is passed, and `mandatory: true` indicates, `--revision` has to be passed when `--tag` is passed.